File: C/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int blocks;
    //Check if input is a +ve Z b/w 1 and 8
    do
    {
        blocks = get_int("Height: ");
    }
    while (blocks < 1 || blocks > 8);

    // row == i
    for (int i = 1; i <= blocks; i++)
    {
        //layer is equal to blocks-row
        for (int j = 0; j < blocks - i; j++)
        {
            printf(" ");
        }
        for (int k = 0; k < i; k++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: songs/Python/sentimental-mario-less/mario.py
import cs50


# main function
def main():
    height = get_int("Height: ", 1, 8)
    half_pyramid(height)


# A variation of the get_int function
def get_int(request, min, max):
    while True:
        value = cs50.get_int(f"{request}")
        if value <= max and value >= min:
            break
    return value


# makes a half_pyramid based on input
def half_pyramid(value):
    for i in range(1, value + 1):
        print(" " * (value - i) + "#" * i)


if __name__ == "__main__":
    main()